packages/cra-template-typescript/template/src/index.tsx

```diff
@@ -1,14 +1,16 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
+import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 
-ReactDOM.render(
+const root = ReactDOM.createRoot(
```

Comment:
Im also getting warnings in the default app.test.js file since it uses the render method from react testing library. Not sure if that needs a similar fix but in case you know better than i do i thought i'd point it out. Thanks so much for this PR i think it'll be really crucial going forward

Reply:
Hey @jbenjoy2, the RTL changes will be done with #12223